python_modules/libraries/dagster-dbt/dagster_dbt/core/utils.py

```diff
@@ -282,3 +283,20 @@ def parse_manifest(path: str, target_path: str = DEFAULT_DBT_TARGET_PATH) -> Map
             return json.load(file)
     except FileNotFoundError:
         raise DagsterDbtCliOutputsNotFoundError(path=manifest_path)
+
+
+def get_future_completion_state_or_err(futures: List[Union[Future, Any]]) -> bool:
```

Comment:
Could we instead use https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.wait with some timeout for similar functionality?

Reply:
Probably, but I think we'd still have to do some custom logic around bookkeeping on event log/Future order. Right now the waiting/timeout is handled in the main loop in `stream` since we just want to wait on the next incomplete event log entry's `Future` (so we can yield it to user code). This way we don't have to worry about yielding things in the wrong order.